src/core/qgsdataprovider.h

```diff
@@ -373,11 +373,15 @@ class CORE_EXPORT QgsDataProvider : public QObject
      * a concept of stack of calls that must be handled by the provider. Only the first
      * call to enterUpdateMode() will really turn update mode on.
      *
+     * Certain providers implicitly enter update mode when committing changes directly to
+     * the provider and specify implicit = true accordingly. For all other cases,
+     * implicit should be false.
+     *
```

Comment:
That doesn't make sense to me to have this flag at the API level. If someone calls enterUpdateMode(), by definition that's explicit !
If you need that flag internally in the OGR provider use a method internal to the provider, but exposing this in the wild is just confusing

Reply:
Fair point

Reply:
Addressed in d7eb1a8